lib/countly.js

```diff
@@ -2956,9 +2958,11 @@
                 }
                 // fallback on error
                 xhr.onreadystatechange = function() {
-                    if (this.readyState === 4 && this.status >= 200 && this.status < 300) {
-                        if (typeof callback === 'function') {
-                            callback(false, params, this.responseText);
+                    //check if operations done, status code is acceptable and the response is JSON 
+                    if (this.readyState === 4 && this.status >= 200 && this.status < 300 && this.responseType === "json") {
+                        //check is callback is a function and response includes a result field
+                        if (typeof callback === 'function' && JSON.stringify(this.response).includes("result")) {
```

Comment:
Performing "JSON.stringify(this.response)" on each response to convert it to a string seems wasteful. Is there no other way to investigate the response without doing a string search?